### ISSUE TITLE
Don't require a translators comment for "foo 100% bar"

### ISSUE THREAD

**Issue by ocean90** [WordPress-Coding-Standards/WordPress-Coding-Standards#776]
```bash
$ echo "<?php __( 'foo 100% bar', 'domain' );" | phpcs --standard=wordpress-core

FILE: STDIN
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 1 | WARNING | A gettext call containing placeholders was found, but
   |         | was not accompanied by a "translators:" comment on the
   |         | line above to clarify the meaning of the placeholders.
----------------------------------------------------------------------

Time: 62ms; Memory: 4Mb
```

Relates to #742.

**Comment by jrfnl** [comment]
Hi @ocean90 Nice case.

I've had a look and would like a second opinion.

First off, `% b` is a valid placeholder (binary number with a space as a padding character). For it not be recognized as a placeholder, `%%` should be used to get a literal `%`.
Which brings me to the real issue (IMHO), that any check which deals with placeholders should possibly first check if the `gettext` call is in a `(s)printf()` call.

The problem with that is, of course, that the `(s)printf()` might not be tightly coupled to the `gettext` call. 

So looking at this, I see three possible solutions:
* Start throwing warnings if placeholders are found in a text string in a `gettext` call when it is not tightly coupled with a `(s)printf()` call. This has the potential of causing a lot of annoyance and risking the I18n sniff from being excluded from rulesets, though that risk could possibly be mitigated with a whitelist comment.
* Don't throw any messages which relate to finding placeholders in a text string in a `gettext` call when it is not tightly coupled with a `(s)printf()` call. This would largely diminish the usefulness of the I18n sniff, so would definitely not have my preference.
* Make the regex twice as big and only recognize the optional padding character if there is also a width specifier in the placeholder. This could potentially mean that some valid placeholders would no longer be recognized as such.

Opinions ?

**Comment by jrfnl** [comment]
As an initial solution direction, I've worked out option 3 to a PR: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/777
